android/src/main/java/com/snowplowanalytics/react/util/EventUtil.java

```diff
@@ -37,165 +39,174 @@ public static SelfDescribingJson createSelfDescribingJson(ReadableMap json) {
     }
 
     public static Structured createStructuredEvent(ReadableMap argmap) {
-        Structured.Builder eventBuilder = Structured.builder()
-            .category(argmap.getString("category"))
-            .action(argmap.getString("action"));
+        Structured event = new Structured(
+                Objects.requireNonNull(argmap.getString("category"), "category can't be null"),
```

Comment:
Just this question (applies for all the `requireNonNull`'s added): why we add this extra check, since we already check/validate in RN side, for example using this [here](https://github.com/snowplow/snowplow-react-native-tracker/blob/809d74f50607f751283ee5b60c4f9b8d501ee5ca/src/events.ts#L101)?

Reply:
Yeah, you're right, it's not really necessary. I was just trying to resolve the warnings that I was getting in Android Studio because they were annoying me. I guess it doesn't hurt to double check although it is better that we check sooner on the RN side.